packages/ragbits-core/src/ragbits/core/vector_stores/_cli.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+from dataclasses import dataclass
+from pathlib import Path
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+
+from ragbits.cli import cli_state, print_output
+from ragbits.cli.state import OutputType
+from ragbits.core.config import core_config
+from ragbits.core.embeddings.base import Embeddings
+from ragbits.core.utils.config_handling import InvalidConfigError
+from ragbits.core.vector_stores.base import VectorStore, VectorStoreOptions
+
+vector_stores_app = typer.Typer(no_args_is_help=True)
+
+
+@dataclass
+class CLIState:
+    vector_store: VectorStore | None = None
+
+
+state: CLIState = CLIState()
+
+
+@vector_stores_app.callback()
+def common_args(
+    factory_path: str | None = None,
+    yaml_path: str | None = None,
+) -> None:
+    try:
+        state.vector_store = VectorStore.subclass_from_defaults(
+            core_config,
+            factory_path_override=factory_path,
+            yaml_path_override=Path.cwd() / yaml_path if yaml_path else None,
+        )
+    except InvalidConfigError as e:
+        Console(stderr=True).print(e)
+        raise typer.Exit(1) from e
+
+
+@vector_stores_app.command(name="list")
+def list_entries(limit: int = 10, offset: int = 0) -> None:
+    """
+    List all objects in the chosen vector store.
+    """
+
+    async def run() -> None:
```

Comment:
why do we use nested function here instead running vector store methods with `asyncio.run` ? i see no concurrency between task within this command

Reply:
This is simply my preference - this way you can use here the same syntax and conventions as with any other asynchronous code in the project. It's the same reason why in our example scripts you usually run `asyncio.run` on the entire `main` function instead of using it only with specific lines inside of `main`. Of course both would work.